src/main/java/org/betonquest/betonquest/objectives/VariableObjective.java

```diff
@@ -98,6 +99,17 @@ public String getProperty(final String name, final Profile profile) {
         return value == null ? "" : value;
     }
 
+    /**
+     * Gets the currently stored variables.
+     *
+     * @param profile the {@link Profile} of the player
+     * @return an unmodifiable copy of the active variable values
+     */
+    public @Nullable Map<String, String> getProperties(final Profile profile) {
+        final VariableData profileData = (VariableData) dataMap.get(profile);
+        return profileData == null ? null : Map.copyOf(profileData.variables);
+    }
```

Comment:
Documentation and code disagree on what is returned: unmodifiable map vs. copy.

You can just wrap the original map in `Collections.unmodifiableMap()` instead:
```suggestion
    /**
     * Gets the currently stored variables.
     *
     * @param profile the {@link Profile} of the player
     * @return the profile's variables as unmodifiable map; or null if the objective is not active for the profile
     */
    public @Nullable Map<String, String> getProperties(final Profile profile) {
        final VariableData profileData = (VariableData) dataMap.get(profile);
        return profileData == null ? null : Collections.unmodifiableMap(profileData.variables);
    }
```